index.js

```diff
@@ -29,6 +29,19 @@ var instructions = [
 ];
 
 /**
+ * Extract protocol information from a URL
+ * Correctly extracts protocols with/without double slash ("//");
```

Comment:
Could you JSDoc comment this function? So it documented what it returns, and what the function expects to receive?